core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -259,10 +259,12 @@ var plugins = map[string]plugin{
 		RMNCrypto:           func(lggr logger.Logger) cciptypes.RMNCrypto { return ccipevm.NewEVMRMNCrypto(lggr) },
 	},
 	chainsel.FamilySolana: {
-		CommitPluginCodec:   ccipsolana.NewCommitPluginCodecV1(),
-		ExecutePluginCodec:  ccipsolana.NewExecutePluginCodecV1(),
-		ExtraArgsCodec:      ccipcommon.NewExtraDataCodec(),
-		MessageHasher:       func(lggr logger.Logger) cciptypes.MessageHasher { return ccipsolana.NewMessageHasherV1(lggr) },
+		CommitPluginCodec:  ccipsolana.NewCommitPluginCodecV1(),
+		ExecutePluginCodec: ccipsolana.NewExecutePluginCodecV1(ccipcommon.NewExtraDataCodec()),
+		ExtraArgsCodec:     ccipcommon.NewExtraDataCodec(),
+		MessageHasher: func(lggr logger.Logger) cciptypes.MessageHasher {
+			return ccipsolana.NewMessageHasherV1(lggr, ccipcommon.NewExtraDataCodec())
```

Comment:
nit: rather than create 2 instantiations of ExtraDataCodec, 1 for member ExtraArgsCodec, and other for member MessageHasher, I'd say, create just 1 and share it.

Reply:
I think the other instantiation goes away completely when we remove it from the ccip reader and delete the decoded arg types.